verifier/src/identity.rs

```diff
@@ -56,8 +60,10 @@ impl TrustedMrEnclaveIdentity {
         J: IntoIterator<Item = &'a HA>,
         HA: ToString + 'a + ?Sized,
     {
+        let mut local_mr_enclave = [0u8; 32];
+        local_mr_enclave.copy_from_slice(mr_enclave.as_ref());
         Self {
-            mr_enclave: *mr_enclave,
+            mr_enclave: local_mr_enclave,
```

Comment:
I'm contemplating removing the serde stuff from this type. as the back and forth between the primitive and the `MrEnclave` type seems error prone.
There is some argument to having a separation between the on wire type and the type used in code logic, but like most things I'm guessing "it depends"

Reply:
I'm probably missing something, but why can't the `mr_enclave` field be of type `MrEnclave` instead of a byte array?

Reply:
It goes back to the serde stuff
MrEnclave supports serde, but not `FromHex`

Reply:
Whats stopping it from supporting `FromHex`?

Reply:
Nothing, just need to do it.  
I'm fine with it, if that's the path we want to take, it shouldn't be hard to do.

Reply:
Cool, I don't have a strong preference, this is all easy to back out of if needed. Just wanted to understand better whats going on. Thanks!